veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,272 @@
+
+
+#### Pros
+
+- Improves the current user experience
+- Will work for hotplugging interfaces using bridge and SR-IOV bindings
+- Keeps the integration with the dynamic-networks-controller
+
+#### Cons
+ - Will not work for non-migratable VMs when SR-IOV NICs are involved or when the dynamic-networks-controller is not deployed 
+ - Performs a migration for every hotplug / hotunplug when the dynamic-networks-controller is not deployed
+ - Requires a minimal API change
+ - Requires the cluster-admin to configure the RolloutStrategy to `LiveUpdate`
+
+## API Examples
+
+<!--
+Tangible API examples used for discussion
+-->
+
+## Alternatives
+
+### Using (only) the dynamic-networks-controller
+
+The dynamic-networks-controller is a controller, running as a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/),
+listening to changes on Pods and invoking the [Multus v4 CNI](https://github.com/k8snetworkplumbingwg/multus-cni).
+
+It enables NIC hotplug / hotunplug operations in place, e.g. without creating another pod.
+A known limitation of the controller above is that it cannot hotplug interfaces that require invocation of
+[Kubernetes device plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/),
+for example: SR-IOV network interfaces.
+
+#### Pros
+
+- Improved user experience
+- No need for migration when attaching/detaching interfaces using bridge binding
+- Works on non-migratable VMs
+- Does not require changes to kubevirt/kubevirt
+
+#### Cons
+
+- The dynamic-networks-controller is an external component that needs to be deployed and managed
+- Not applicable for hotplugging SR-IOV interfaces
+
+### Providing Namespace Admins the required RBAC Rules
+
+Cluster admins could provide Namespace admins the permissions to create VMIM objects.
+The documentation will be updated to reflect the fact that there are additional RBAC rules missing for namespace
+admins to be able to perform NIC hotplug / unplug.
+
+#### Pros
+
+- Does not require code changes
+
+#### Cons
+
+- Losses the benefits of limiting VMIM creation to cluster admins
+- Requires cluster-admins to take action
+
+## Scalability
+
+N/A.
+
+## Update/Rollback Compatibility
+
+Users will have to configure their KubeVirt configuration similarly to CPU and memory hotplug in order to enable the automatic
+migration.
+
+## Functional Testing Approach
+
+The migration-based hotplug / hotunplug tests will be adjusted to remove the manual VMIM object creation.
+The in-place hotplug / hotunplug tests will be adjusted to assert that a migration did not occur.
+
+## Implementation Phases
+
+1. Support automatic migration on NIC hotplug / hotunplug (without dynamic-networks-controller).
+2. Fix the redundant `RestartRequired` condition added to VMs when NIC hotplug / hotunplug are performed.
+
+## Feature lifecycle Phases
+
+### GA
+Planned for v1.6.
```

Comment:
I'm personally not against this but if folks are you could provide a configurable policy to switch between the existing manual and this new automatic behaviour, the latter also sitting behind a FG that we could graduate.

Reply:
The goal of initiating automatic migration was (among others) to address the issue introduced by the VMIM hardening while not providing namespace admins the permissions to create VMIM.
It is basically a bug fix.

Additionally, both NIC hotplug and LiveUpdate rollout strategy are GAed, so it will be confusing IMO to have their integration as non GA.
Also, which code paths should be protected by a FG? in which configuration would the e2e test execute? etc.

Reply:
yes, I tend to agree that we can do without it.